speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,471 @@
+
+        super().__init__(
+            source=source,
+            save_path=save_path,
+            freeze=freeze,
+            cache_dir=cache_dir,
+            device=device,
+            *args,
+            **kwargs,
+        )
+
+        self.sampling_rate = sampling_rate
+        self.encoder_only = encoder_only
+        self.freeze_encoder = freeze_encoder
+        self.freeze_feature_extractor = freeze_feature_extractor
+        self.output_attentions = output_attentions
+        self.output_all_hiddens = output_all_hiddens
+        self.generation_config = None
+
+        self.load_tokenizer(source=source, **kwargs)
+
+        config = SpeechT5Config.from_pretrained(
+            source, cache_dir=cache_dir, **kwargs
+        )
+        self.config = self.override_config(config)
+
+        if self.encoder_only:
+            logger.warning(
+                f"{self.__class__.__name__}: Using an encoder-only SpeechT5 for speech to text."
+            )
+            del self.model.speecht5.decoder
+            del self.model.text_decoder_postnet
+
+        if self.freeze_feature_extractor:
+            self.model.freeze_feature_encoder()
+
+    def forward(self, wav, decoder_input_ids=None):
+        """
+        One forward step of the SpeechT5 model ** using the manually implemented methods ** by calling
+        the `forward_encoder()` and `forward_decoder()` methods.
+
+        Arguments
+        ----------
+            wav (raw waveform to be fed to the model.): torch.Tensor or numpy.ndarray
+            decoder_input_ids (If using a decoder, those would be the tokenized text targets.): list
+
+        Returns
+        -------
+            Encoder-only: Either the encoder hidden layers (last or all internal stacked if `self.output_all_hiddens` is `True`)
+            Encoder-Decoder: The encoder hidden states (see encoder-only), the logits, and the attentions
+        """
+
+        encoder_output = self.forward_encoder(wav)
+        if self.output_all_hiddens:
+            encoder_output = encoder_output[-1]
+
+        if self.encoder_only:
+            return encoder_output
+
+        # In the same fashion as mBART:
+        # We replace 0 elements by pax_idx as pad_idx of SpeechT5 is 1
+        decoder_input_ids = custom_padding(
+            x=decoder_input_ids,
+            org_pad=0,
+            custom_pad=self.model.speecht5.decoder.config.pad_token_id,
+        )
+
+        if self.freeze:
+            with torch.no_grad():
+                logits, attentions = self.forward_decoder(
+                    audio_features=encoder_output,
+                    decoder_input_ids=decoder_input_ids,
+                )
+        else:
+            logits, attentions = self.forward_decoder(
+                audio_features=encoder_output,
+                decoder_input_ids=decoder_input_ids,
+            )
+
+        return encoder_output, logits, attentions
+
+    def forward_encoder(self, wav) -> torch.Tensor:
```

Comment:
can you please add a docstring here?:

Reply:
I kept getting a flake error somehow because of the docstring. I was saying that the wav param was not in the docstring even though it was. Strange...

Reply:
Ok can you please add it back? I can have a look at why it is failing

Reply:
Done. Even by running flake 8 locally I get this error:
```bash
.\speechbrain\lobes\models\huggingface_transformers\speecht5.py:189:1: DOC101 Method `SpeechT5ForASR.forward_encoder`: Docstring contains fewer arguments than in function signature. 
.\speechbrain\lobes\models\huggingface_transformers\speecht5.py:189:1: DOC109 Method `SpeechT5ForASR.forward_encoder`: The option `--arg-type-hints-in-docstring` is `True` but there are no type hints in the docstring arg list 
.\speechbrain\lobes\models\huggingface_transformers\speecht5.py:189:1: DOC103 Method `SpeechT5ForASR.forward_encoder`: Docstring arguments are different from function arguments. (Or could be other formatting issues: https://jsh9.github.io/pydoclint/violation_codes.html#notes-on-doc103 ). Arguments in the function signature but not in the docstring: [wav: ].
```